.github/workflows/ur-build-hw.yml

```diff
@@ -0,0 +1,132 @@
+name: UR - Build adapters on HW
+
+on:
+  workflow_call:
```

Comment:
We prefer helper workflows like this one to have an additional workflow_dispatch trigger for testing/experimenting, e.g. https://github.com/intel/llvm/blob/822cf9b028726b730bb4156a0704acd8201f69aa/.github/workflows/sycl-linux-run-tests.yml#L106